src/components/ImageView/ShowImageModal.vue

```diff
@@ -0,0 +1,76 @@
+<template>
+	<NcModal v-if="show"
+		size="large"
+		:title="currentImage.basename"
+		:outTransition="true"
+		:hasNext="true"
+		:hasPrevious="true"
+		closeButtonContained="false"
+		dark="true"
+		@next="showNextImage"
+		@previous="showPreviousImage"
+		@close="$emit('close')">
+		<div class="modal__content">
+			<img :src="currentImage.source">
+		</div>
+	</NcModal>
+</template>
+
+<script>
+import NcModal from '@nextcloud/vue/dist/Components/NcModal'
+
+export default {
+	name: 'ShowImageModal',
+	components: {
+		NcModal,
+	},
+	props: {
+		images: {
+			type: Array,
+			default() {
+				return []
+			},
+			validator(imagesList) {
+				return (imagesList.length === 0)
+					? true
+					: imagesList.findIndex(({ basename, source }) => !(basename && source)) !== -1
+			},
+		},
+		startIndex: {
```

Comment:
```suggestion
		imageIndex: {
```
Because `startIndex` suggests that it is definitive and that the prop is not reactive but it is! :+1: